Yen/learnNavigation/app/src/main/java/com/example/learnnavigation/viewmodel/BackgroundImageViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.learnnavigation.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.learnnavigation.data.api.ApiService
+import com.example.learnnavigation.model.Demo
+import com.example.learnnavigation.model.SuggestedSearches
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class BackgroundImageViewModel : ViewModel() {
+
+    private val _suggestedSearches = MutableLiveData<List<SuggestedSearches>>()
+    val suggestedSearches: LiveData<List<SuggestedSearches>> = _suggestedSearches
+
+    fun fetchDataFromApi() {
```

Comment:
fun traceErrorException(throwable: Throwable?): ApiError {

    return when (throwable) {

        is HttpException -> {
            when (throwable.code()) {
                400 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.BAD_REQUEST
                )
                401 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.UNAUTHORIZED
                )
                403 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.FORBIDDEN
                )
                404 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.NOT_FOUND
                )
                405 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.METHOD_NOT_ALLOWED
                )
                409 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.CONFLICT
                )
                500 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.INTERNAL_SERVER_ERROR
                )
                else -> ApiError(
                    UNKNOWN_ERROR_MESSAGE,
                    0,
                    ApiError.ErrorStatus.UNKNOWN_ERROR
                )
            }
        }

        is SocketTimeoutException -> {
            ApiError(throwable.message, ApiError.ErrorStatus.TIMEOUT)
        }

        is IOException -> {
            ApiError(throwable.message, ApiError.ErrorStatus.NO_CONNECTION)
        }

        else -> ApiError(UNKNOWN_ERROR_MESSAGE, 0, ApiError.ErrorStatus.UNKNOWN_ERROR)
    }

đại khái tạo 1 Extension Func xử lý connect và lỗi kiểu như ví dụ trên, success code từ 200..299 còn lại 300..500 là lỗi